### PR TITLE
upgrade Golang to 1.21

### PR DESCRIPTION
**Pull request by glours**
**What I did**
Bump Golang to version `1.21`

**Related issue**
N/A

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
image

### CODE REVIEW

**Review by ndeloof — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-ARG GO_VERSION=1.20.7
+ARG GO_VERSION=1.21
```
Comment (milas) [inline Dockerfile:18]:
```suggestion
ARG GO_VERSION=1.21.0
```

We've generally been pinning to the patch as well, right? 

That said...I'm fine with using the rolling tag (`1.21`) if you want, despite the `Host` issues recently, Go patch upgrades are generally very safe IMO, so up to you!

**Review by milas — APPROVED** [review]

**Comment by cpuguy83** [comment]
Why?
Please add reasons _why_ this is changed not just that it's changed, that's plain to see from the code that there's a change.

**Comment by ndeloof** [comment]
@cpuguy83 why would we want to keep an older version ?

**Comment by cpuguy83** [comment]
Bumping the go.mod means everyone who compiles it has to bump the go version.
What would you not support a still in-support go version?

Anyway, the point is not to say "you shouldn't have done this" but "what's the reason for the decision?".

**Comment by ndeloof** [comment]
> everyone who compiles it has to bump the go version

obviously! But if you exclude a few Linux distribution maintainers who have to compile on their own, the many others just consume Docker Compose as a pre-built binary and aren't affected.

> what's the reason for the decision?

because we want to rely on latest available and can benefits potential optimizations or new constructs it offers. We don't wait to see a coding opportunity to bump version.

**Pull request merged**